homeassistant/components/ubee/device_tracker.py

```diff
@@ -14,47 +14,47 @@
     CONF_HOST, CONF_PASSWORD, CONF_USERNAME)
 import homeassistant.helpers.config_validation as cv
 
-REQUIREMENTS = ['pyubee==0.2']
+REQUIREMENTS = ['pyubee==0.4']
 
 _LOGGER = logging.getLogger(__name__)
 
+CONF_MODEL = 'model'
+
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Required(CONF_HOST): cv.string,
     vol.Required(CONF_PASSWORD): cv.string,
     vol.Required(CONF_USERNAME): cv.string,
+    vol.Optional(CONF_MODEL, default='detect'): cv.string,
```

Comment:
Maybe make sense to limit supported model to what it is:
```
LIST_SUPPORTED_MODEL_BY_LIB = ['H23423', 'O23423']
DEFAULT_MODEL = "detect"
...
vol.Optional(CONF_MODEL, default=DEFAULT_MODEL): vol.All(cv.string, vol.In(LIST_SUPPORTED))
```

Reply:
Thanks @pvizeli .
There is list of supported models in [pyubee](https://github.com/mzdrale/pyubee), so I could just import it:

```python
from pyubee import SUPPORTED_MODELS
```

but if I understood correctly, requirement is to import lib component is depending on out in function. So this wouldn't be valid:

```python
import logging
import voluptuous as vol

from homeassistant.components.device_tracker import (
    DOMAIN, PLATFORM_SCHEMA, DeviceScanner)
from homeassistant.const import (
    CONF_HOST, CONF_PASSWORD, CONF_USERNAME)
import homeassistant.helpers.config_validation as cv

from pyubee import SUPPORTED_MODELS

REQUIREMENTS = ['pyubee==0.4']

_LOGGER = logging.getLogger(__name__)

CONF_MODEL = 'model'
DEFAULT_MODEL = 'detect'

PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
    vol.Required(CONF_HOST): cv.string,
    vol.Required(CONF_PASSWORD): cv.string,
    vol.Required(CONF_USERNAME): cv.string,
    vol.Optional(CONF_MODEL, default=DEFAULT_MODEL): vol.All(cv.string, vol.In(SUPPORTED_MODELS))
})
``` 

I wanted to avoid to update list of supported models in this component each time new model is added to [pyubee](https://github.com/mzdrale/pyubee), if possible, but if it's requirement, I will do it.

Reply:
You can make your own function:
```
def check_type(value):
   import xy
   if not value in fff:
      raise vol.Invalid()
   return value
...
): vol.All(cv.string, check_typte)
```

Reply:
Changed code as you suggested.